app/web/modules/discrepancies/discrepancies.http.js

```diff
@@ -0,0 +1,33 @@
+const logger = require('./../../../lib/logger')
+
+const { Connector } = require('./../../../lib/pay-request')
+const { wrapAsyncErrorHandlers } = require('./../../../lib/routes')
+
+const search = async function searchTransaction (req, res) {
+  res.render('discrepancies/report')
+}
+
+const getDiscrepancyReport = async function searchTransaction (req, res) {
+  const chargeIds = req.body.search_string.split(',')
+  const comparisons = await Connector.getGatewayComparisons(chargeIds)
+
+  const interpretedComparisons = comparisons.map(comparison => {
+    const canBeResolved = comparison.gatewayExternalStatus !== comparison.payExternalStatus;
```

Comment:
Is there any reason this can't just be
```js 
comparison.canBeResolved = comparison.gatewayExternalStatus !== comparison.payExternalStatus
return comparison
```

Reply:
no. I thought it was clearer the way I had it. But it isn't really, is it

Reply:
oh, and I am now allergic to mutating objects in place

Reply:
re: immutable
```js 
comparisons.map(comparison => Object.assign({} , comparison, { canBeResolved: comparison.gatewayExternalStatus !== comparison.payExternalStatus }))
```